tasks/windows.ps1

```diff
@@ -0,0 +1,41 @@
+[CmdletBinding()]
+Param(
+  [Parameter(Mandatory = $True)] [String] $master,
+  [Parameter(Mandatory = $False)] [String] $cacert_content,
+  [Parameter(Mandatory = $False)] [String] $certname,
+  [Parameter(Mandatory = $False)] [String] $dns_alt_names
+)
+
+Function Validate-Parameter {
+  Param ($param)
+  if ($param -contains "'") {
+    Throw 'Single-quote is not allowed in arguments'
+  }
+}
+
+Validate-Parameter($certname)
+Validate-Parameter($alt_names)
+
+if ($certname.Length -eq 0) {
+  $certname_arg = "agent:certname='$certname' "
+}
+else {
+  $fqdn = [System.Net.Dns]::GetHostByName(($env:computerName)).Hostname 
```

Comment:
This is a reasonable start, but this approach can be problematic, especially in environments like AWS and Azure.